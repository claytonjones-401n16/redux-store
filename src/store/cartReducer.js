export const initState = {
  cartCount: 0,
  cartContents: [],
}

export default (state = initState, action) => {
  let newState = { ...state };

  switch (action.type) {
    case 'ADD_TO_CART':
      newState.cartContents = [...newState.cartContents, action.payload];
      newState.cartCount++;
      break;
    case 'REMOVE_FROM_CART':
      for (let i = 0; i < newState.cartContents.length; i++) {
        if (action.payload.name === newState.cartContents[i].name) {
          newState.cartContents.splice(i, 1);
          newState.cartContents = [...newState.cartContents];

          newState.cartCount--;
          break;
        }
      }
      break;
    default:
      break;
  }

  return newState;
}