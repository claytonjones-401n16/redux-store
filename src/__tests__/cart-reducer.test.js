import cartReducer, {initState} from '../store/cartReducer';

describe('cart reducer', () => {
  it('should return init state', () => {
    expect(cartReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle ADD_TO_CART', () => {
    expect(cartReducer(initState,
      {
        type: 'ADD_TO_CART',
        payload: {name: 'testProduct'}
      })
    ).toMatchSnapshot();
  });

  it('should handle REMOVE_FROM_CART', () => {
    expect(cartReducer(
      {cartContents: [{name: 'test'}], cartCount: 1},
      {
        type: 'REMOVE_FROM_CART',
        payload: 'test'
      }
    )).toMatchSnapshot();
  })
})