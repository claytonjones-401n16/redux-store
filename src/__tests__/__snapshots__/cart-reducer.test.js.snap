// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart reducer should handle ADD_TO_CART 1`] = `
Object {
  "cartContents": Array [
    Object {
      "name": "testProduct",
    },
  ],
  "cartCount": 1,
}
`;

exports[`cart reducer should handle REMOVE_FROM_CART 1`] = `
Object {
  "cartContents": Array [
    Object {
      "name": "test",
    },
  ],
  "cartCount": 1,
}
`;

exports[`cart reducer should return init state 1`] = `
Object {
  "cartContents": Array [],
  "cartCount": 0,
}
`;
